components/eventing-controller/controllers/subscription_controller.go

```diff
@@ -14,24 +24,343 @@ import (
+	}
+
+	// Sync the BEB Subscription with the Subscription CR
+	if statusChangedForBeb, err := r.syncBEBSubscription(subscription, &result, ctx, log); err != nil {
+		log.Error(err, "error while syncing BEB subscription")
+		return ctrl.Result{}, err
+	} else {
+		statusChanged = statusChanged || statusChangedForBeb
+	}
+
+	if r.isInDeletion(subscription) {
+		// Remove finalizers
+		if err := r.removeFinalizer(subscription, ctx, log); err != nil {
+			return ctrl.Result{}, err
+		}
+		result.Requeue = false
+		return result, nil
+	}
+
+	// Save the subscription status if it was changed
+	if statusChanged {
+		if err := r.Status().Update(ctx, subscription); err != nil {
+			log.Error(err, "Update subscription status failed")
+			return ctrl.Result{}, err
+		}
+		result.Requeue = true
+	}
+
+	return result, nil
+}
+
+// syncFinalizer sets the finalizer in the Subscription
+func (r *SubscriptionReconciler) syncFinalizer(subscription *eventingv1alpha1.Subscription, result *ctrl.Result, ctx context.Context, logger logr.Logger) error {
+	// Check if finalizer is already set
+	if r.isFinalizerSet(subscription) {
+		return nil
+	}
+	if err := r.addFinalizer(subscription, ctx, logger); err != nil {
+		return err
+	}
+	result.Requeue = true
+	return nil
+}
+
+// syncBEBSubscription it returns
+func (r *SubscriptionReconciler) syncBEBSubscription(subscription *eventingv1alpha1.Subscription,
+	result *ctrl.Result, ctx context.Context, logger logr.Logger) (bool, error) {
+	logger.Info("Syncing subscription with BEB")
+
+	r.bebClient.Initialize()
+
+	// if object is marked for deletion, we need to delete the BEB subscription
+	if r.isInDeletion(subscription) {
+		return false, r.deleteBEBSubscription(subscription, logger, ctx)
+	}
+
+	var statusChanged bool
+	var err error
+	if statusChanged, err = r.bebClient.SyncBebSubscription(subscription); err != nil {
+		logger.Error(err, "Update BEB subscription failed")
+		return false, err
+	}
+
+	if !subscription.Status.IsConditionSubscribed() {
+		condition := eventingv1alpha1.MakeCondition(eventingv1alpha1.ConditionSubscribed, eventingv1alpha1.ConditionReasonSubscriptionCreated, corev1.ConditionTrue)
+		if err := r.replaceStatusCondition(subscription, condition); err != nil {
+			return statusChanged, err
+		}
+		statusChanged = true
+	}
+
+	statusChangedAtCheck, retry, errTimeout := r.checkStatusActive(subscription)
+	statusChanged = statusChanged || statusChangedAtCheck
+	if errTimeout != nil {
+		logger.Error(errTimeout, "Timeout at retry")
+		result.Requeue = false
+		return statusChanged, errTimeout
+	}
+	if retry {
+		logger.Info("Wait for subscription to be active", "name:", subscription.Name, "status:", subscription.Status.EmsSubscriptionStatus.SubscriptionStatus)
+		condition := eventingv1alpha1.MakeCondition(eventingv1alpha1.ConditionSubscriptionActive, eventingv1alpha1.ConditionReasonSubscriptionNotActive, corev1.ConditionFalse)
+		result.RequeueAfter = time.Second * 1
```

Comment:
It is retried only for 30s, after that it will be processed like an K8S error. Usually the "retry" is instantaneously and it should be used after that a resource was saved to etcd, to obtain get the new value of the resource